graviton/abi_strategy.py

```diff
@@ -140,11 +140,16 @@ def address_logic(x):
                 )
             )
 
-        waterfall = OrderedDict(
+        waterfall: Dict[
+            Union[type[abi.ABIType], str], Callable[..., PyTypes]
+        ] = OrderedDict(
             [
                 (abi.UfixedType, not_implemented),
                 (abi.BoolType, lambda x: not x),
-                (abi.UintType, lambda x: (1 << self.abi_type.bit_size) - 1 - x),
+                (
+                    abi.UintType,
```

Comment:
The fact that we need to cast to the type that is actually matched in the waterfall gets me thinking that there ought to be a more fundamental fix by the processor of this `OrderedDict`

Reply:
yea, `waterfall: OrderedDict` does not specify the relation between key and item, so I have to do the type cast here... maybe we can take as is, but refactor waterfall later...

Reply:
ok, I am refactoring out map into `mutate_for_roundtrip` to match case by case the type against `self.abi_type`.

Reply:
should be refactored in commit [eccf0c8](https://github.com/algorand/graviton/pull/46/commits/eccf0c821bab275b045933289a2ef94a177541e7) for `map` method.